### PR TITLE
[feat]: [SRM-13767]: Enable ELK MTLS

### PR DESCRIPTION
**Pull request by AbhijithHarness**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46087)

### CODE REVIEW

File: 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java [hunk 1]
```diff
@@ -915,6 +915,19 @@ public ScheduledExecutorService healthMonitorExecutor() {
...
+  public DataCollectionDSLService dataCollectionDSLService() throws KeyManagerBuilderException {
+    if (StringUtils.isNotEmpty(configuration.getClientCertificateFilePath())
+        && StringUtils.isNotEmpty(configuration.getClientCertificateKeyFilePath())) {
+      KeyManager keyManager = new X509KeyManagerBuilder()
+                                  .withClientCertificateFromFile(this.configuration.getClientCertificateFilePath(),
+                                      this.configuration.getClientCertificateKeyFilePath())
+                                  .build();
+      return new DataCollectionServiceImpl(keyManager);
+    }
+    return new DataCollectionServiceImpl();
```
Comment (dhruv-harness) [inline 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java:928]: nit: We can add an else block for easier readability.

**Review by dhruv-harness — APPROVED** [review]

**Review by deepak-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46087)* status: 4 file(s) to review. 1 change requests. 

___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/elkconnector/ELKCapabilityHelper.java` line 28 at r1](https://reviewable.io/reviews/harness/harness-core/46087#-NSErpnU12wye3OsPuCL:-NSErpnU12wye3OsPuCM:b5qq2es) ([raw file](https://github.com/harness/harness-core/blob/ccb18fe0fcc12d0b9338c3145876e1bbf97691fd/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/connector/elkconnector/ELKCapabilityHelper.java#L28)):*
> ```Java
>     // Checking capability for basic http call as if mTLS is enabled, the connection to actual elk server will hang up
>     capabilityList.add(HttpConnectionExecutionCapabilityGenerator.buildHttpConnectionExecutionCapability(
>         "https://www.elastic.co/", maskingEvaluator));
> ```

Can't user ask for MTLS on in-house deployed elk server?

**Review by deepak-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46087)* status: :shipit: complete! ✓

File: 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java (hunk 1 above)
Comment (n00bitax) [reply 260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java:928]: what happens if only one file is there, or both are abscent?

**Review by n00bitax — APPROVED** [review]

**Comment by AbhijithHarness** [comment]
trigger ut8

**Review by AbhijithHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46087)* status: 2. unresolved discussions. 

___
*[`260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java` line 928 at r1](https://reviewable.io/reviews/harness/harness-core/46087#-NSEii9w4ktS2IKihMdW-r1-928:-NSFxNa4DuR1A9P3cNLb:b-c790p5) ([raw file](https://github.com/harness/harness-core/blob/ccb18fe0fcc12d0b9338c3145876e1bbf97691fd/260-delegate/src/main/java/io/harness/delegate/app/modules/DelegateModule.java#L928)):*
<details><summary><i>Previously, n00bitax (Ansuman Satapathy) wrote…</i></summary><blockquote>

nit: will it error out here if the file is not a valid cert or the key is not valid for cert here?
</blockquote></details>

1 file being absent is same as both files being absent,  as we cannot do anything with Just 1 file.
Yes it will error out, if certificates are invalid.

**Comment by AbhijithHarness** [comment]
trigger ut8

**Pull request merged**